src/pathchk/pathchk.rs

```diff
@@ -0,0 +1,229 @@
+#![allow(unused_must_use)] // because we of writeln!
+#![crate_name = "uu_pathchk"]
+
+/*
+ * This file is part of the uutils coreutils package.
+ *
+ * (c) Inokentiy Babushkin <inokentiy.babushkin@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE file
+ * that was distributed with this source code.
+ */
+
+extern crate getopts;
+extern crate libc;
+
+#[macro_use]
+extern crate uucore;
+
+use getopts::Options;
+use std::fs;
+use std::io::{Write, ErrorKind};
+
+// operating mode
+enum Mode {
+    Default, // use filesystem to determine information and limits
+    Basic,   // check basic compatibility with POSIX
+    Extra,   // check for leading dashes and empty names
+    Both,    // a combination of `Basic` and `Extra`
+    Help,    // show help
+    Version  // show version information
+}
+
+static NAME: &'static str = "pathchk";
+static VERSION: &'static str = env!("CARGO_PKG_VERSION");
+
+// a few global constants as used in the GNU implememntation
```

Comment:
Did you pull these constants from the GNU source code or did you get them some other way?